File: javascript/test/Creational/Builder.js
(function() {
  define(['Creational/Builder/BostonSoupBuffetBuilder', 'Creational/Builder/HonoluluSoupBuffetBuilder'], function(BostonSoupBuffetBuilder, HonoluluSoupBuffetBuilder) {
    'use strict';
    var CreateSoupBuffet;
    CreateSoupBuffet = function(soupBuffetBuilder) {
      soupBuffetBuilder.buildSoupBuffet();
      soupBuffetBuilder.setSoupBuffetName();
      soupBuffetBuilder.buildChickenSoup();
      soupBuffetBuilder.buildClamChowder();
      soupBuffetBuilder.buildFishChowder();
      soupBuffetBuilder.buildMinnestrone();
      soupBuffetBuilder.buildPastaFazul();
      soupBuffetBuilder.buildTofuSoup();
      soupBuffetBuilder.buildVegetableSoup();
      return soupBuffetBuilder.getSoupBuffet();
    };
    return describe('Test Builder Pattern', function() {
      var bostonsoupBuffet, honoluluSoupBuffet;
      it('builder files should be loaded', function() {
        expect(BostonSoupBuffetBuilder).toBeDefined();
        return expect(HonoluluSoupBuffetBuilder).toBeDefined();
      });
      bostonsoupBuffet = CreateSoupBuffet(new BostonSoupBuffetBuilder());
      it("Testing Boston soup buffett", function() {
        return expect("At the " + (bostonsoupBuffet.getSoupBuffetName()) + " today's soup is " + (bostonsoupBuffet.getTodaysSoups())).toEqual("At the Boston Soup Buffet today's soup is  Today's Soups!  , Chicken Soup: ,ChickenSoup, Clam Chowder: ,QuahogChowder, Fish Chowder: ,ScrodFishChowder, Minnestrone: ,Minnestrone, Pasta Fazul: ,Pasta Fazul, Tofu Soup: ,Tofu Soup, Vegetable Soup: ,Vegetable Soup");
      });
      honoluluSoupBuffet = CreateSoupBuffet(new HonoluluSoupBuffetBuilder());
      return it('Testing Honolulu soup buffett', function() {
        return expect("At the " + (honoluluSoupBuffet.getSoupBuffetName()) + " today's soup is " + (honoluluSoupBuffet.getTodaysSoups())).toEqual("At the Honolulu Soup Buffet today's soup is  Today's Soups!  , Chicken Soup: ,ChickenSoup, Clam Chowder: ,PacificClamChowder, Fish Chowder: ,OpakapakaFishChowder, Minnestrone: ,Minnestrone, Pasta Fazul: ,Pasta Fazul, Tofu Soup: ,Tofu Soup, Vegetable Soup: ,Vegetable Soup");
      });
    });
  });

}).call(this);

File: java/t/test/Creational/Builder.java
package test.Creational;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import common.*;
import Creational.Builder.*;


//TestSoupBuffetBuilder.java - Testing the builder

public class Builder {

   public SoupBuffet CreateSoupBuffet(
     SoupBuffetBuilder soupBuffetBuilder) { 
        soupBuffetBuilder.buildSoupBuffet();
        
        soupBuffetBuilder.setSoupBuffetName();
        
        soupBuffetBuilder.buildChickenSoup();
        soupBuffetBuilder.buildClamChowder();
        soupBuffetBuilder.buildFishChowder();    
        soupBuffetBuilder.buildMinnestrone();
        soupBuffetBuilder.buildPastaFazul();
        soupBuffetBuilder.buildTofuSoup();
        soupBuffetBuilder.buildVegetableSoup();
        
        return soupBuffetBuilder.getSoupBuffet();
   }     
   
	@Test 
   public void thetester() {
       
       SoupBuffet bostonSoupBuffet = CreateSoupBuffet(new BostonSoupBuffetBuilder());
       org.junit.Assert.assertEquals("Boston soup buffet builder test",
			"At the Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + bostonSoupBuffet.getSoupBuffetName() + bostonSoupBuffet.getTodaysSoups()
			);

       SoupBuffet honoluluSoupBuffet = 
         CreateSoupBuffet(new HonoluluSoupBuffetBuilder());
       org.junit.Assert.assertEquals("Honolulu soup buffet builder test",
			"At the Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + honoluluSoupBuffet.getSoupBuffetName() + honoluluSoupBuffet.getTodaysSoups()
		);
   }
}      
